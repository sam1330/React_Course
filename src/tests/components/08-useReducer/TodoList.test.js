import { shallow } from "enzyme";
import React from "react";

import TodoList from "../../../components/08-useReducer/components/TodoList";
import { demoTodos } from "../../fixtures/demoTodos";

describe('<TodoList /> tests', () => {

  const handleDelete = jest.fn();
  const handleToggle = jest.fn();


  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test('should render the component', () => {

    expect(wrapper).toMatchSnapshot();

  });

  test('should render <TodoListItem /> with Its properties', () => {

    expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
    expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
    expect(wrapper.find('TodoListItem').at(0).prop('handleToggle')).toEqual(expect.any(Function));

  });
});