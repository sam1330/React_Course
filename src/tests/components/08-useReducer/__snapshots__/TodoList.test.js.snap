// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> tests should render the component 1`] = `
<Fragment>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      key="1"
      todo={
        Object {
          "desc": "Learn React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      key="2"
      todo={
        Object {
          "desc": "Learn Node",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</Fragment>
`;
